### PR TITLE
Feat: Improve from datasets

### PR DESCRIPTION
**Pull request by dcfidalgo**
This PR improves a bit the `Dataset*.from_datasets` mechanics. Instead of running x-times over the entire dataset for parsing x columns, we do the parsing of all columns in one go over the dataset.

- improves performance
- makes the ignore columns message less confusing

It also provides a "proper" fix to support the new datasets 2.3.0 version (not the push_to_hub issue, but the one when renaming a column).

### CODE REVIEW

File: src/rubrix/client/datasets.py [hunk 1]
```diff
@@ -153,88 +155,146 @@ def _to_datasets_dict(self) -> Dict:
...
 
-    @classmethod
-    def _from_datasets(cls, dataset: "datasets.Dataset") -> "Dataset":
-        """Helper method to create a Rubrix Dataset from a datasets Dataset.
+    @staticmethod
+    def _join_datasets_columns_and_delete(
+        row: Dict[str, Any], columns: List[str]
+    ) -> Dict[str, Any]:
+        """Joins columns of a `datasets.Dataset` row into a dict, and deletes the single columns.
+        UPDATES TE ROW!
```
Comment (frascuchon) [inline src/rubrix/client/datasets.py]: ?

**Review by frascuchon — DISMISSED** [review]

File: src/rubrix/client/datasets.py (hunk 1 above)
Comment (dcfidalgo) [inline src/rubrix/client/datasets.py]:
```suggestion

        Updates the ``row`` dictionary!
        
```

File: src/rubrix/client/datasets.py [hunk 2]
```diff
@@ -153,88 +155,148 @@ def _to_datasets_dict(self) -> Dict:
...
-        """Helper method to create a Rubrix Dataset from a datasets Dataset.
+    @staticmethod
+    def _join_datasets_columns_and_delete(
+        row: Dict[str, Any], columns: List[str]
+    ) -> Dict[str, Any]:
+        """Joins columns of a `datasets.Dataset` row into a dict, and deletes the single columns.
 
-        Must be implemented by the child class.
+        Updates the ``row`` dictionary!
+
```
Comment (dcfidalgo) [inline src/rubrix/client/datasets.py]:
```suggestion
```

File: src/rubrix/client/datasets.py (hunk 1 above)
Comment (dcfidalgo) [reply src/rubrix/client/datasets.py]: Typo, wanted to explicitly state that the method updates the ``row`` dictionary.

**Pull request merged**